// @atlaskit/tokens/dist/cjs/constants.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TOKEN_NOT_FOUND_CSS_VAR = exports.THEME_NAME_MAP = exports.THEME_DATA_ATTRIBUTE = exports.THEMES = exports.DEFAULT_THEME = exports.CSS_VAR_FULL = exports.CSS_PREFIX = void 0;
var THEMES = ['light', 'dark', 'legacy-light', 'legacy-dark'];
exports.THEMES = THEMES;
var THEME_DATA_ATTRIBUTE = 'data-theme';
exports.THEME_DATA_ATTRIBUTE = THEME_DATA_ATTRIBUTE;
var DEFAULT_THEME = 'light';
exports.DEFAULT_THEME = DEFAULT_THEME;
var CSS_PREFIX = 'ds';
exports.CSS_PREFIX = CSS_PREFIX;
var CSS_VAR_FULL = ['opacity']; // Maps the longer theme name to a shorthand used in css/code

exports.CSS_VAR_FULL = CSS_VAR_FULL;
var THEME_NAME_MAP = {
  'atlassian-light': 'light',
  'atlassian-dark': 'dark',
  'atlassian-legacy-light': 'legacy-light',
  'atlassian-legacy-dark': 'legacy-dark'
};
exports.THEME_NAME_MAP = THEME_NAME_MAP;
var TOKEN_NOT_FOUND_CSS_VAR = "--".concat(CSS_PREFIX, "-token-not-found");
exports.TOKEN_NOT_FOUND_CSS_VAR = TOKEN_NOT_FOUND_CSS_VAR;
